src/inference/__tests__/inference.test.ts

```diff
@@ -65,3 +66,146 @@ describe('Inference Class: embed', () => {
     }
   });
 });
+
+describe('Inference Class: rerank', () => {
+  let inference: Inference;
+
+  beforeEach(() => {
+    const config: PineconeConfiguration = { apiKey: 'test-api-key' };
+    const infApi = inferenceOperationsBuilder(config);
+    inference = new Inference(infApi);
+  });
+
+  test('Throws error if no documents are passed', async () => {
+    const rerankingModel = 'test-model';
+    const myQuery = 'test-query';
+    const expectedError = new PineconeArgumentError(
+      'You must pass at least one document to rerank'
+    );
+    try {
+      await inference.rerank(rerankingModel, myQuery, []);
+    } catch (error) {
+      expect(error).toEqual(expectedError);
+    }
+  });
+
+  test('Throws error if list of dicts is passed for docs, but does not contain `text` key', async () => {
```

Comment:
nit: They're objects not dicts in javascript, technically. Sorry I know that's pedantic. I would just replace all dict/dictionary references to objects.

Reply:
Will do!